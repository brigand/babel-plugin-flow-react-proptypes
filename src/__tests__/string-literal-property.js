const babel = require('babel-core');
const content = `
export type T = {
  'read-only': true,
  regular: true
};
`;

it('string-literal-property', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['env', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
