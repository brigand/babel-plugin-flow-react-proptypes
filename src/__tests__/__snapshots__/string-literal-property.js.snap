// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`string-literal-property 1`] = `
"'use strict';

var babelPluginFlowReactPropTypes_proptype_T = {
  \\"read-only\\": require('prop-types').oneOf([true]).isRequired,
  regular: require('prop-types').oneOf([true]).isRequired
};
if (typeof exports !== 'undefined') Object.defineProperty(exports, 'babelPluginFlowReactPropTypes_proptype_T', {
  value: babelPluginFlowReactPropTypes_proptype_T,
  configurable: true,
  enumerable: true
});"
`;
